### PR TITLE
letsencrypt: Fix joker.com domain configuration

### PR DESCRIPTION
**Pull request by 4nx**
Hi there,

I recently added this https://github.com/home-assistant/addons/pull/3642, but was now realising that there is already a new version 2.1.0 and I was missing the domain variable in the credential (/data/dnsapikey) file. This PR should fix this both and will update CHANGELOGS and DOCS accordingly. 

## Summary by CodeRabbit

- **New Features**
  - Added support for Joker.com DNS challenges in Let's Encrypt.
  - Introduced configuration options for Joker DNS: `joker_username`, `joker_password`, and `joker_domain`.

- **Bug Fixes**
  - Fixed an issue with missing domain configuration for Joker.com DNS challenges in version 5.0.23.

### CODE REVIEW

**Comment by agners** [comment]
> I recently added this #3642, but was now realising that there is already a new version 2.1.0 and I was missing the domain variable in the credential (/data/dnsapikey) file. This PR should fix this both and will update CHANGELOGS and DOCS accordingly.

The 2.1.0 release doesn't seem to have made it onto PyPI though :cry: 

https://pypi.org/project/certbot-dns-joker/

**Comment by 4nx** [comment]
> > I recently added this #3642, but was now realising that there is already a new version 2.1.0 and I was missing the domain variable in the credential (/data/dnsapikey) file. This PR should fix this both and will update CHANGELOGS and DOCS accordingly.
> 
> The 2.1.0 release doesn't seem to have made it onto PyPI though 😢
> 
> https://pypi.org/project/certbot-dns-joker/

Ok now I know why I was confused, because my first PR was based on informationens from PyPI and so I wondered why I missed the 2.1.0 after looking at Github. I will change the commit accordingly back to 1.1.0 but will keep the changes regarding the domain variable because this seems to be needed anyway.

File: letsencrypt/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## 5.0.23
+
+- Upgrade certbot-dns-joker to 2.1.0 and fix missing domain
```
Comment (agners) [inline letsencrypt/CHANGELOG.md:5]:
```suggestion
- Fix missing domain for Joker DNS challenge
```

File: letsencrypt/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## 5.0.23
+
+- Fix missing domain for joker.com DNS challenge
```
[the remark this answers is not in the document]
Comment (agners) [reply letsencrypt/CHANGELOG.md:5]: I guess you are right :smile: 

```suggestion
- Fix missing domain configuration for joker.com DNS challenge
```

**Review by agners — APPROVED** [review]

**Pull request merged**